docs/src/devdocs/reference_cells.md

```diff
@@ -1,17 +1,263 @@
 # Reference cells
 
 The reference cells are used to i) define grid cells, ii) define shape functions, and iii)
-define quadrature rules. The numbering of vertices, edges, faces are visualized below. See also
+define quadrature rules. The numbering of vertices, edges, faces are visualized below. See
+also
 [`FerriteViz.elementinfo`](https://ferrite-fem.github.io/FerriteViz.jl/dev/api/#FerriteViz.elementinfo).
 
+## Numbering and identification of entities
+
+The local numbering of vertices, edges, and faces, on the reference cells, specifies, for
+example, the cell node order, and the order in which interpolations distribute their DoFs.
+It is important for internal consistency and correctness that the the same convention is
+used everywhere. The convention adopted by Ferrite.jl is documented below for each reference
+cell.
+
+**Edges** are identified by the 2-tuple of vertices it connects, starting with the smallest
+vertex number, i.e. ``(v_i, v_j)`` where ``i < j``.
+
+**Faces** are identified by the n-tuple of vertices constructing the face in anti-clockwise
+order (viewing the face from the outside of the reference cell), starting with the smallest
+vertex number, i.e. ``(v_i, v_j, v_k, ...)`` where ``i < j, i < k, i < ...``.
```

Comment:
Actually not right? We only require that the first node is the smallest.

Reply:
IIUC then we always sort these fully for faces, see https://github.com/Ferrite-FEM/Ferrite.jl/blob/master/src/Dofs/DofHandler.jl#L729-L735

Reply:
What are $v_i,v_j,...$ then? I am also not confident that the faces for all reference elements are counter-clockwise ordered. At least we had in the past the scenario that not all were, but I cannot remember if we changed this. The faces are ordered in a way such that they give us a proper entity of lower dimension.

Also note that the edges are always not ordered as above either.

Reply:
Does this ordering really matter that much? 

I guess the 2d edges are just made such that `face = first.(e1, e2, ...)`, but for the edge I think one only have to care about this when setting up vector interpolations, such that the positive direction is in the same direction as the edge, see e.g. [here](https://github.com/Ferrite-FEM/Ferrite.jl/blob/714d67c483d292f8244b53d990e0ac9ae0df2c54/src/interpolations.jl#L1717-L1740)). During cellvalues mapping, we simply need to [reverse ](https://github.com/Ferrite-FEM/Ferrite.jl/blob/714d67c483d292f8244b53d990e0ac9ae0df2c54/src/FEValues/FunctionValues.jl#L198) if `edges(cell)[edgenr][1] > edges(cell)[edgenr][2]`.

And in 3d (haven't implemented Hdiv for that yet where it matters for me), but didn't you say @fredrikekre that we should consider having something like

```julia
function face_direction(cell, facenr)
    face = faces(cell)[facenr]
    N = length(face)
    imin = argmin(face
    return face[mod1(imin-1, N)] > face[mod1(imin+1, N)] ? 1 : -1
end
```
(Do we already use this somewhere?)
I'm too tired now to think if this will work independent of local ordering, but I think it will? 

To me, the convention of starting with the lowest local vertex index is reasonable. Not sure if we have to require counter-clockwise, but sounds like a good thing to enforce in any case. I looked through the docs preview, and at least there all reference shapes have are ordered counter-clockwise. However, with moving all reference information to this new file, I would suggest that we test the ordering, 

I think something like this
```julia 
x = reference_coordinates(...)
xc = sum(x)/lenght(x) # refshape center
for fk in faces(refshape)
    xfc = sum(x[i] for i in fk)/length(fk) # face center
    n = cross(x[fk[i+1]] - x[fk[i]], x[fk[i+2]] - x[fk[i+1]])
    @test cdot(n, xfc - xc) > 0 # Normal is pointing out
end
```
should do it?

Reply:
> IIUC then we always sort these fully for faces, see https://github.com/Ferrite-FEM/Ferrite.jl/blob/master/src/Dofs/DofHandler.jl#L729-L735

But the convention would be used to populate that `SurfaceOrientationInfo` I think?

Reply:
> I am also not confident that the faces for all reference elements are counter-clockwise ordered. At least we had in the past the scenario that not all were, but I cannot remember if we changed this. The faces are ordered in a way such that they give us a proper entity of lower dimension.

I am pretty sure this is true and tested. cellnodes for 2d shapes are counter clockwise, and faces for 3d shapes should be so too, starting with the lowest index.

Reply:
> Also note that the edges are always not ordered as described above either 

Yea I think I wrote this when I had the idea to renumber just like defelement. Not sure what issues switching the edges would cause though.

Reply:
At least https://github.com/Ferrite-FEM/Ferrite.jl/blob/bdd5b41f1485aef8274953a4d2dd361582d69e94/src/Grid/grid.jl#L212  face 3 for prism is not starting from the lowest index. Regarding the orientation, does this https://github.com/Ferrite-FEM/Ferrite.jl/blob/bdd5b41f1485aef8274953a4d2dd361582d69e94/test/test_interpolations.jl#L164-L166 test the counter clockwise orientation?

Reply:
> test the counter clockwise orientation?

Nice, yes, but for the interpolation numbering AFAIU, 
https://github.com/Ferrite-FEM/Ferrite.jl/blob/bdd5b41f1485aef8274953a4d2dd361582d69e94/src/interpolations.jl#L881C50-L881C57

the refshape numbering should match for linear interpolations,
https://github.com/Ferrite-FEM/Ferrite.jl/blob/bdd5b41f1485aef8274953a4d2dd361582d69e94/src/Grid/grid.jl#L200-L203

It only tests for the first 3 though, so for quad faces, it would be possible to have ordering like 
```julia
3  4
1  2
```
pass.

Reply:
> > test the counter clockwise orientation?
> 
> Nice, yes, but for the interpolation numbering AFAIU, 
> 
> https://github.com/Ferrite-FEM/Ferrite.jl/blob/bdd5b41f1485aef8274953a4d2dd361582d69e94/src/interpolations.jl#L881-L881
> 
> the refshape numbering should match for linear interpolations,
> 
By our convention via `geometric_interpolation` the interpolation numbering here *must* match the numbering in the reference cell, right?

> https://github.com/Ferrite-FEM/Ferrite.jl/blob/bdd5b41f1485aef8274953a4d2dd361582d69e94/src/Grid/grid.jl#L200-L203
> 
> It only tests for the first 3 though, so for quad faces, it would be possible to have ordering like
> 
> ```julia
> 3  4
> 1  2
> ```
> 
> pass.